Jenkinsfile

```
@@ -5,8 +5,8 @@ def use_newer_jenkins = random.nextBoolean() // Use newer Jenkins on one build b
 
 // build recommended configurations
 subsetConfiguration = [ [ jdk: '8',  platform: 'windows', jenkins: null                      ],
-                        [ jdk: '8',  platform: 'linux',   jenkins: !use_newer_jenkins ? '2.176.3' : '2.164.1', javaLevel: '8' ],
-                        [ jdk: '11', platform: 'linux',   jenkins:  use_newer_jenkins ? '2.176.3' : '2.164.1', javaLevel: '8' ]
+                        [ jdk: '8',  platform: 'linux',   jenkins: !use_newer_jenkins ? '2.204.6' : '2.222.1', javaLevel: '8' ],
+                        [ jdk: '11', platform: 'linux',   jenkins:  use_newer_jenkins ? '2.204.6' : '2.222.1', javaLevel: '8' ]
```

I know you're only updating the jenkins version, but I'd like to understand this two lines. Wouldn't be the same as
```suggestion
                        [ jdk: '8',  platform: 'linux',   jenkins: '2.222.1', javaLevel: '8' ],
                        [ jdk: '11', platform: 'linux',   jenkins:  '2.204.6', javaLevel: '8' ]
```
?

---

Good question.  I will add a comment explaining that logic.

The random boolean `use_newer_jenkins` will test Java 8 on Linux with Jenkins 2.204.6 roughly 50% of the time and with Jenkins 2.222.1 roughly 50% of the time.  The negation of that random boolean will test Java 11 on Linux with the version of Jenkins which was not used in the Java 8 test.

If the random boolean is true, then Java 8  will test 2.204.6 and Java 11 will test 2.222.1.

If the random boolean is false, then Java 8 will test 2.222.1 and Java 11 will test 2.204.6..

---

See comments added in 105b7e7f1a18a9c68e33f3cb4e341c1d273c509c and 92f3c2de5fce739e84b1d044e0d7adc2f6824162